penty/pentypes/builtins.py

```diff
@@ -472,21 +479,246 @@ def set_init(elts_ty=None):
+    if not issubclass(self_ty, set):
+        raise TypeError
+    return _Set[self_ty.__args__[0].copy()]
+
+def set_difference(self_ty, *other_tys):
+    from penty.penty import Types
+    if not issubclass(self_ty, set):
+        raise TypeError
+    for other_ty in other_tys:
+        if '__iter__' not in Types[other_ty]:
+            raise TypeError
+    return _Set[self_ty.__args__[0].copy()]
+
+def set_difference_update(self_ty, *other_tys):
+    if not issubclass(self_ty, set):
+        raise TypeError
+    set_difference(self_ty, *other_tys)
+    return _Cst[None]
+
+def set_discard(self_ty, elt_ty):
+    if not issubclass(self_ty, set):
+        raise TypeError
+    if not self_ty.__args__[0]:
+        raise TypeError
+    return _Cst[None]
+
+def set_intersection(self_ty, *other_tys):
+    from penty.penty import Types
+    if not issubclass(self_ty, set):
+        raise TypeError
+    intersection_tys = self_ty.__args__[0].copy()
+    for other_ty in other_tys:
+        if '__iter__' not in Types[other_ty]:
+            raise TypeError
+        iter_tys = Types[other_ty]['__iter__'](other_ty)
+        if not isinstance(iter_tys, set):
+            iter_tys = {iter_tys}
+        for iter_ty in iter_tys:
+            if '__next__' not in Types[iter_ty]:
+                raise TypeError
+            value_ty = Types[iter_ty]['__next__'](iter_ty)
+            intersection_tys.add(value_ty)
+    return _Set[intersection_tys]
+
+def set_intersection_update(self_ty, *other_tys):
+    if not issubclass(self_ty, set):
+        raise TypeError
+    updated = set_intersection(self_ty, *other_tys)
+    self_ty.__args__[0].update(updated.__args__[0])
+    return _Cst[None]
+
+def set_isdisjoint(self_ty, other_ty):
+    from penty.penty import Types
+    if not issubclass(self_ty, set):
+        raise TypeError
+    if not self_ty.__args__[0] or not other_ty.__args__[0]:
+        return _Cst[True]
+    if '__iter__' not in Types[other_ty]:
+        raise TypeError
+    return bool
+
+def set_issubset(self_ty, other_ty):
+    from penty.penty import Types
+    if not issubclass(self_ty, set):
+        raise TypeError
+    if not self_ty.__args__[0]:
+        return _Cst[True]
+    if '__iter__' not in Types[other_ty]:
+        raise TypeError
+    return bool
+
+def set_issuperset(self_ty, other_ty):
+    if not issubclass(self_ty, set):
+        raise TypeError
+    if not self_ty.__args__[0] and not other_ty.__args__[0]:
+        return _Cst[True]
+    return set_issubset(self_ty, other_ty)
+
+def set_pop(self_ty):
+    if not issubclass(self_ty, set):
+        raise TypeError
+    if not self_ty.__args__[0]:
+        raise TypeError
+    return self_ty.__args__[0].copy()
+
+def set_remove(self_ty, value_ty):
+    if not issubclass(self_ty, set):
+        raise TypeError
+    if not self_ty.__args__[0]:
+        raise TypeError
+    return _Cst[None]
+
+def set_symmetric_difference(self_ty, other_ty):
+    if not issubclass(self_ty, set):
```

Comment:
This check is redoundant with the one done in `set_intersection`.

You could simplify this with:

```python
 def set_symmetric_difference(self_ty, other_ty):
    return set_intersection(self_ty, other_ty)
```

Reply:
and the same remark holds for other methods:

- `set_symmetric_difference_update`
- `set_union`
- `set_update`